runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -139,6 +160,29 @@ private static void setupSlf4jReporting(MetricRegistry metrics) {
         slf4jReporter.start(metricsLogInterval, TimeUnit.SECONDS);
     }
 
+    // If enabled, setup csv reporting
+    private static void setupCsvReporting(MetricRegistry metrics) {
+        if (!metricsCsvReportingEnabled) return;
+
+        final File directory = new File(metricsCsvFolder);
+
+        if (!directory.isDirectory() ||
+            !directory.exists() ||
+            !directory.canWrite()) {
+            log.warn("Provided CSV directory : {} doesn't exist, isn't a directory, or not accessible " +
+                    "for writes. Disabling CSV Reporting.");
+            metricsCsvReportingEnabled = false;
+            return;
+        }
+
+        CsvReporter csvReporter = CsvReporter.forRegistry(metrics)
+                .convertDurationsTo(TimeUnit.MICROSECONDS)
+                .convertRatesTo(TimeUnit.SECONDS)
+                .filter(ADDRESS_SPACE_FILTER)
+                .build(directory);
```

Comment:
It creates a CSV per metric in the provided folder given its flag is on. They are geared towards easy trouble shooting and are generated when the CSV flag is on. Rather than scraping the logs, this provides enables technical and non-technical users to simply open the CSV files in excel or similar tools and analyze the metrics and correlations between them.